the sego.hide:message reiases: MP3_Encoder.py:1368: OverflowError

stego = Steganography(quiet=True)
    stego.hide_message(audio_path, output_path, message)

.venv/lib/python3.11/site-packages/mp3stego/steganography.py:157: in hide_message
    too_long = self.__encode(wav_file_path, output_file_path, hide=True, bitrate=self.__bitrate, quiet=self.quiet,
.venv/lib/python3.11/site-packages/mp3stego/steganography.py:50: in __encode
    return self.__encoder.encode(quiet=quiet)
.venv/lib/python3.11/site-packages/mp3stego/encoder/encoder.py:45: in encode
    self.__encoder.encode()
.venv/lib/python3.11/site-packages/mp3stego/encoder/MP3_Encoder.py:607: in encode
    written, data = self.__encode_buffer_internal()
.venv/lib/python3.11/site-packages/mp3stego/encoder/MP3_Encoder.py:644: in __encode_buffer_internal
    self.__format_bitstream()
.venv/lib/python3.11/site-packages/mp3stego/encoder/MP3_Encoder.py:1278: in __format_bitstream
    self.__encode_main_data()
.venv/lib/python3.11/site-packages/mp3stego/encoder/MP3_Encoder.py:1359: in __encode_main_data
    self.__huffman_code_bits(gr, ch)
.venv/lib/python3.11/site-packages/mp3stego/encoder/MP3_Encoder.py:1440: in __huffman_code_bits
    self.__put_bits(~0, 32)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <mp3stego.encoder.MP3_Encoder.MP3Encoder object at 0x7c42f4118950>
val = -1, N = 32

    def __put_bits(self, val, N):
        """
        write N bits into the bit stream.
    
        :param val: value to write into the buffer
        :param N: number of bits of val
        """
>       val = np.uint32(val)
E       OverflowError: Python integer -1 out of bounds for uint32

.venv/lib/python3.11/site-packages/mp3stego/encoder/MP3_Encoder.py:1368: OverflowError
----------------------------- Captured stderr call -----------------------------
decoding:  88%|████████▊ | 616905/703648 [00:34<00:04, 17802.98it/s]
encoding:   0%|          | 0/738 [00:00<?, ?it/s]

--- a/.gitignore
+++ b/.gitignore
@@ -4,3 +4,7 @@
 /build/
 /dist/
 /mp3stego_lib.egg-info/
+__pycache__
+cleared.mp3
+out.mp3
+reveal.txt

--- a/mp3stego/encoder/MP3_Encoder.py
+++ b/mp3stego/encoder/MP3_Encoder.py
@@ -8,6 +8,7 @@ from tqdm import tqdm
 from mp3stego.encoder import tables
 from mp3stego.encoder import util
 from mp3stego.encoder.WAV_Reader import WavReader
+from mp3stego.utils import safe_uint32
 
 NUM_OF_HUF_TABLES = 32
 
@@ -1437,11 +1438,11 @@ class MP3Encoder:
 
             # Due to the nature of the Huffman code tables, we will pad with ones * /
             while stuffing_words:
-                self.__put_bits(~0, 32)
+                self.__put_bits(safe_uint32(~0), 32)
                 stuffing_words -= 1
 
             if remaining_bits:
-                self.__put_bits(np.uint32(np.left_shift(np.uint32(np.uint64(1)), np.int(remaining_bits)) - 1),
+                self.__put_bits(np.uint32(np.left_shift(np.uint32(np.uint64(1)), np.int_(remaining_bits)) - 1),
                                 remaining_bits)
 
     def __huffman_code(self, table_select, x, y):

--- a/mp3stego/utils.py
+++ b/mp3stego/utils.py
@@ -1,0 +1,7 @@
+def safe_uint32(value):
+    """
+    Converts negative integers to their unsigned equivalent within the uint32 range.
+    """
+    if isinstance(value, int) and value < 0:
+        return value & 0xFFFFFFFF  # Wrap negative values into uint32 range
+    return value

